src/java.security.jgss/share/classes/sun/security/krb5/internal/KDCRep.java

```diff
@@ -138,7 +138,7 @@ protected void init(DerValue encoding, int req_type)
                         " req type is " + req_type);
 
                 System.out.println(">>> KDCRep: Message in bytes is =>");
-                byte[] dataBytes = encoding.getDataBytes();
+                byte[] dataBytes = encoding.getData().toByteArray();
```

Comment:
This change must differ from later versions because `toByteArray()` might return `null` (if there are no bytes in encoding). I expect an exception to be thrown by
```java
        der = encoding.getData().getDerValue();
```
on line 159 below, but enabling debugging should replace that with a `NullPointerException` on line 143.

Reply:
The `toByteArray()` will only throw a `NullPointerException` if the buffer is null. But this cannot happen since the constructor of the original `DerValue` would throw an `IOException`.

Am I missing some path that could cause that `NullPointerException`?

Reply:
And the constructor of `DerValue` never gets a null argument, since the `KdcComm` class, where the byte array is created, checks and throws an `IOException` itself, if the response byte array is null.

Reply:
Notice, I didn't say that `toByteArray()` would throw an exception - I said it might return `null`. See [`DerInputBuffer.toByteArray()`](https://github.com/ibmruntimes/openj9-openjdk-jdk11/blob/openj9/src/java.base/share/classes/sun/security/util/DerInputBuffer.java#L76).

On line 143, the expression `dataBytes.length` will throw `NullPointerException` if `dataBytes` is `null`. We must not allow that to happen; we need to reach line 157 where the `Asn1Exception` is thrown.

Reply:
Oh, I see. You mean if the original byte array supplied to the `DerValue` is not null, but is empty, in which case the `DerInputBuffer.toByteArray()` will return null.

So should I just catch the exception and move on, or should I throw the `Asn1Exception` right away?

Reply:
Changed it to catch the exception and just print another message saying the reply is empty.